### PR TITLE
DAOS-5359 control: Clean up JSON output in dmg

### PR DESCRIPTION
**Pull request by mjmac**
Errors should be included in the JSON output. Adds
a new test to verify that all commands can produce
JSON results.

### CODE REVIEW

File: src/control/cmd/dmg/pool.go [hunk 1]
```diff
@@ -370,12 +417,13 @@ func (c *PoolQueryCmd) Execute(args []string) error {
 
 	ctx := context.Background()
 	resp, err := control.PoolQuery(ctx, c.ctlInvoker, req)
-	if err != nil {
-		return errors.Wrap(err, "pool query failed")
-	}
 
 	if c.jsonOutputEnabled() {
```
Comment (tanabarr) [inline src/control/cmd/dmg/pool.go:421]: just an idea , but couldn't we add `var wrap string` at the beginning of the `Execute()` methods and `defer func() {if c.jsonOutputEnabled() {return c.errorJSON(err)}; return errors.Wrap(err, wrap)}`, setting wrap var before we return in the body of the `Execute()` method (not setting wrap var will basically make the defer a noop). it might be possible to integrate that behaviour into the json Cmd type to further reduce boilerplate.

File: src/control/cmd/dmg/storage.go [hunk 2]
```diff
@@ -219,6 +226,28 @@ func (cmd *storageFormatCmd) shouldReformatSystem(ctx context.Context) (bool, er
 	return false, nil
 }
 
+func (cmd *storageFormatCmd) systemReformat(ctx context.Context) error {
```
Comment (tanabarr) [inline src/control/cmd/dmg/storage.go:229]: whilst a good change, this reorganisation is a bit out of scope for the PR

File: src/control/cmd/dmg/pool.go (hunk 1 above)
Comment (mjmac) [reply src/control/cmd/dmg/pool.go:421]: I considered something like that and a few other approaches, but eventually settled on what is there now because it's explicit. I don't love the extra typing but at least it's easy to follow. I need to get this landed in order to enable the use of dmg in C tests -- we can revisit this in the future if something more clever occurs to us.

File: src/control/cmd/dmg/storage.go [hunk 3]
```diff
@@ -78,6 +80,9 @@ func (cmd *storagePrepareCmd) Execute(args []string) error {
 	}
 
 	if prepScm {
+		if cmd.jsonOutputEnabled() && !cmd.Force {
+			return cmd.errorJSON(errors.New("Cannot use --json without --force"))
+		}
```
Comment (kjacque) [inline src/control/cmd/dmg/storage.go:85]: Shouldn't we check this up at the top before we do anything else? Is there a reason why we wouldn't want to check this when prepping NVMe?

Comment (mjmac) [reply src/control/cmd/dmg/storage.go:85]: Well, the main reason I put it in the prepScm block is that we only prompt for user confirmation when we're preparing SCM (due to the need to reboot).
